### ISSUE TITLE
Bold variants not displayed correctly with Qt

### ISSUE THREAD

**Issue by naoh40** [adobe-fonts/source-han-sans#197]
Not sure if this is a QT bug of a font bug, so I'm going to file this to both.

Description
====
Here are the strange behaviors of all Source Han Sans fonts (or of Qt):

- No issues for GTK apps.
- All Qt programs do not display Source Han Sans variants correctly. ```fc-match``` shows me the correct font names and file names, but when a program wants the bold variant of a Source Han Sans font, for example Source Han Sans CN Bold, what is displayed is always a _faux_ bold version of the font, as is shown in the screenshot.

![bug](https://user-images.githubusercontent.com/38598931/39091548-658aee64-4628-11e8-9e38-6ab92752f461.png)

Here is my observation:
- I moved ```/etc/fonts``` to somewhere else and then tried running Qt programs as the ```root``` user but the problem persists, so this can't be a problem with my config files.
- I tried displaying those characters in a virtual machine with a Kubuntu 18.04 LiveCD, and reproduced the problem successfully.
- It's not only a problem with Source Han Sans *, but also with Noto Sans CJK.

Sorry for the long issue.

**Comment by kenlunde** [comment]
First, _Noto Sans CJK_ was built the same as _Source Han Sans_, and their fonts differ only by their names. Second, if the issue that you described does not occur when using _Source Han Serif_ or the Google-branded _Noto Serif CJK_, then it is a style-linking issue. In other words, Qt apps expect typeface families to be style-linked, or at least expects the **Bold** weight to be style-linked.

We implemented style-linking for _Source Han Serif_, and plan to do so for _Source Han Sans_ as part of the Version 2.000 update. See the second bullet in Issue #181. I provided instructions for building your own style-linked fonts, for those who need them before the Version 2.000 update, in the _COMMANDS.txt_ file in the master branch of the project.
